hollow/src/main/java/com/netflix/hollow/api/producer/AbstractHollowProducer.java

```diff
@@ -807,7 +807,21 @@ private void announce(ProducerListeners listeners, HollowProducer.ReadState read
         if (announcer != null) {
             Status.StageWithStateBuilder status = listeners.fireAnnouncementStart(readState);
             try {
-                announcer.announce(readState.getVersion(), readState.getStateEngine().getHeaderTags());
+                // don't allow producer to enable/disable between validating primary status and then announcing
+                singleProducerEnforcer.lock();
+                try {
+                    if (!singleProducerEnforcer.isPrimary()) {
+                        // This situation can arise when the producer is asked to relinquish primary status mid-cycle.
+                        // If this non-primary producer is allowed to announce then a different producer instance could
+                        // have been running concurrently as primary and that would break the delta chain.
+                        log.log(Level.INFO,
+                                "Fail the announcement because current producer is not primary (aka leader)");
```

Comment:
I _think_ internally we tend to say "lease holder" rather than "leader", not sure which one is more precise though

Reply:
Leader, being more generic, seems more appropriate here from the point of view of the user